compiler/rustc_parse/src/parser/item.rs

```diff
@@ -2658,16 +2658,35 @@ impl<'a> Parser<'a> {
     ///
     /// `vis` represents the visibility that was already parsed, if any. Use
     /// `Visibility::Inherited` when no visibility is known.
+    ///
+    /// If `is_fn_pointer_type`, we error on `const` and `async` qualifiers,
+    /// which are not allowed in function pointer types.
     pub(super) fn parse_fn_front_matter(
         &mut self,
         orig_vis: &Visibility,
         case: Case,
+        is_fn_pointer_type: bool,
```

Comment:
Could you make this an enum? Right now the boolean isn't very clear at the callsite